
<launch>

	<arg name="debug" default="false"/>
  	<arg if="$(arg debug)" name="launch_prefix" value="xterm -rv -e gdb -ex run - args"/>
  	<arg unless="$(arg debug)" name="launch_prefix" value=""/>


	<node name="ar_pose_c1_n0" pkg="ar_pose" type="ar_single" respawn="false"
   	 output="screen">

		<remap from="camera/image_raw" to="camera1/rgb/image" />
		<remap from="camera/camera_info" to="camera1/rgb/camera_info" />
   	 	<param name="marker_pattern" type="string" value="$(find ar_pose)/data/4x4/4x4_19.patt"/>
    		<param name="marker_width" type="double" value="96.0"/>
    		<param name="marker_center_x" type="double" value="0.0"/>
    		<param name="marker_center_y" type="double" value="0.0"/>
    		<param name="threshold" type="int" value="120"/>
    		<param name="use_history" type="bool" value="true"/>
		<param name="marker_frame" value="marker" />
		<param name="publish_tf" value="false" />
		<remap from="ar_pose_marker" to="ar_pose/cam1/cp_0" />

		<param name="rgb_frame_id" value="camera_left" />
  	</node>

	<node name="ar_pose_c2_n0" pkg="ar_pose" type="ar_single" respawn="false"
   	 output="screen">

		<remap from="camera/image_raw" to="camera2/rgb/image" />
		<remap from="camera/camera_info" to="camera2/rgb/camera_info" />
   	 	<param name="marker_pattern" type="string" value="$(find ar_pose)/data/4x4/4x4_19.patt"/>
    		<param name="marker_width" type="double" value="96.0"/>
    		<param name="marker_center_x" type="double" value="0.0"/>
    		<param name="marker_center_y" type="double" value="0.0"/>
    		<param name="threshold" type="int" value="120"/>
    		<param name="use_history" type="bool" value="true"/>
		<param name="marker_frame" value="ar_marker" />
		<param name="publish_tf" value="false" />
		<remap from="ar_pose_marker" to="ar_pose/cam2/cp_0" />

		<param name="rgb_frame_id" value="camera_right" />
  	</node>


	<node name="ar_pose_c1_n1" pkg="ar_pose" type="ar_single" respawn="false"
   	 output="screen">

		<remap from="camera/image_raw" to="camera1/rgb/image" />
		<remap from="camera/camera_info" to="camera1/rgb/camera_info" />
   	 	<param name="marker_pattern" type="string" value="$(find ar_pose)/data/4x4/4x4_33.patt"/>
    		<param name="marker_width" type="double" value="96.0"/>
    		<param name="marker_center_x" type="double" value="0.0"/>
    		<param name="marker_center_y" type="double" value="0.0"/>
    		<param name="threshold" type="int" value="120"/>
    		<param name="use_history" type="bool" value="true"/>
		<param name="marker_frame" value="marker" />
		<param name="publish_tf" value="false" />
		<remap from="ar_pose_marker" to="ar_pose/cam1/cp_1" />

		<param name="rgb_frame_id" value="camera_left" />
  	</node>

	<node name="ar_pose_c2_n1" pkg="ar_pose" type="ar_single" respawn="false"
   	 output="screen">

		<remap from="camera/image_raw" to="camera2/rgb/image" />
		<remap from="camera/camera_info" to="camera2/rgb/camera_info" />
   	 	<param name="marker_pattern" type="string" value="$(find ar_pose)/data/4x4/4x4_33.patt"/>
    		<param name="marker_width" type="double" value="96.0"/>
    		<param name="marker_center_x" type="double" value="0.0"/>
    		<param name="marker_center_y" type="double" value="0.0"/>
    		<param name="threshold" type="int" value="120"/>
    		<param name="use_history" type="bool" value="true"/>
		<param name="marker_frame" value="ar_marker" />
		<param name="publish_tf" value="false" />
		<remap from="ar_pose_marker" to="ar_pose/cam2/cp_1" />

		<param name="rgb_frame_id" value="camera_right" />
  	</node>

	<node name="ar_pose_c1_n2" pkg="ar_pose" type="ar_single" respawn="false"
   	 output="screen">

		<remap from="camera/image_raw" to="camera1/rgb/image" />
		<remap from="camera/camera_info" to="camera1/rgb/camera_info" />
   	 	<param name="marker_pattern" type="string" value="$(find ar_pose)/data/4x4/4x4_62.patt"/>
    		<param name="marker_width" type="double" value="96.0"/>
    		<param name="marker_center_x" type="double" value="0.0"/>
    		<param name="marker_center_y" type="double" value="0.0"/>
    		<param name="threshold" type="int" value="120"/>
    		<param name="use_history" type="bool" value="true"/>
		<param name="marker_frame" value="marker" />
		<param name="publish_tf" value="false" />
		<remap from="ar_pose_marker" to="ar_pose/cam1/cp_2" />

		<param name="rgb_frame_id" value="camera_left" />
  	</node>

	<node name="ar_pose_c2_n2" pkg="ar_pose" type="ar_single" respawn="false"
   	 output="screen">

		<remap from="camera/image_raw" to="camera2/rgb/image" />
		<remap from="camera/camera_info" to="camera2/rgb/camera_info" />
   	 	<param name="marker_pattern" type="string" value="$(find ar_pose)/data/4x4/4x4_62.patt"/>
    		<param name="marker_width" type="double" value="96.0"/>
    		<param name="marker_center_x" type="double" value="0.0"/>
    		<param name="marker_center_y" type="double" value="0.0"/>
    		<param name="threshold" type="int" value="120"/>
    		<param name="use_history" type="bool" value="true"/>
		<param name="marker_frame" value="ar_marker" />
		<param name="publish_tf" value="false" />
		<remap from="ar_pose_marker" to="ar_pose/cam2/cp_2" />

		<param name="rgb_frame_id" value="camera_right" />
  	</node>

	<node name="ar_pose_c1_n3" pkg="ar_pose" type="ar_single" respawn="false"
   	 output="screen">

		<remap from="camera/image_raw" to="camera1/rgb/image" />
		<remap from="camera/camera_info" to="camera1/rgb/camera_info" />
   	 	<param name="marker_pattern" type="string" value="$(find ar_pose)/data/4x4/4x4_70.patt"/>
    		<param name="marker_width" type="double" value="96.0"/>
    		<param name="marker_center_x" type="double" value="0.0"/>
    		<param name="marker_center_y" type="double" value="0.0"/>
    		<param name="threshold" type="int" value="120"/>
    		<param name="use_history" type="bool" value="true"/>
		<param name="marker_frame" value="marker" />
		<param name="publish_tf" value="false" />
		<remap from="ar_pose_marker" to="ar_pose/cam1/cp_3" />

		<param name="rgb_frame_id" value="camera_left" />
  	</node>

	<node name="ar_pose_c2_n3" pkg="ar_pose" type="ar_single" respawn="false"
   	 output="screen">

		<remap from="camera/image_raw" to="camera2/rgb/image" />
		<remap from="camera/camera_info" to="camera2/rgb/camera_info" />
   	 	<param name="marker_pattern" type="string" value="$(find ar_pose)/data/4x4/4x4_70.patt"/>
    		<param name="marker_width" type="double" value="96.0"/>
    		<param name="marker_center_x" type="double" value="0.0"/>
    		<param name="marker_center_y" type="double" value="0.0"/>
    		<param name="threshold" type="int" value="120"/>
    		<param name="use_history" type="bool" value="true"/>
		<param name="marker_frame" value="ar_marker" />
		<param name="publish_tf" value="false" />
		<remap from="ar_pose_marker" to="ar_pose/cam2/cp_3" />

		<param name="rgb_frame_id" value="camera_right" />
  	</node>


	<node name="ar_pose_c1_n4" pkg="ar_pose" type="ar_single" respawn="false"
   	 output="screen">

		<remap from="camera/image_raw" to="camera1/rgb/image" />
		<remap from="camera/camera_info" to="camera1/rgb/camera_info" />
   	 	<param name="marker_pattern" type="string" value="$(find ar_pose)/data/4x4/4x4_80.patt"/>
    		<param name="marker_width" type="double" value="96.0"/>
    		<param name="marker_center_x" type="double" value="0.0"/>
    		<param name="marker_center_y" type="double" value="0.0"/>
    		<param name="threshold" type="int" value="120"/>
    		<param name="use_history" type="bool" value="true"/>
		<param name="marker_frame" value="marker" />
		<param name="publish_tf" value="false" />
		<remap from="ar_pose_marker" to="ar_pose/cam1/cp_4" />

		<param name="rgb_frame_id" value="camera_left" />
  	</node>

	<node name="ar_pose_c2_n4" pkg="ar_pose" type="ar_single" respawn="false"
   	 output="screen">

		<remap from="camera/image_raw" to="camera2/rgb/image" />
		<remap from="camera/camera_info" to="camera2/rgb/camera_info" />
   	 	<param name="marker_pattern" type="string" value="$(find ar_pose)/data/4x4/4x4_80.patt"/>
    		<param name="marker_width" type="double" value="96.0"/>
    		<param name="marker_center_x" type="double" value="0.0"/>
    		<param name="marker_center_y" type="double" value="0.0"/>
    		<param name="threshold" type="int" value="120"/>
    		<param name="use_history" type="bool" value="true"/>
		<param name="marker_frame" value="ar_marker" />
		<param name="publish_tf" value="false" />
		<remap from="ar_pose_marker" to="ar_pose/cam2/cp_4" />

		<param name="rgb_frame_id" value="camera_right" />
  	</node>

	<node name="ar_pose_c1_n5" pkg="ar_pose" type="ar_single" respawn="false"
   	 output="screen">

		<remap from="camera/image_raw" to="camera1/rgb/image" />
		<remap from="camera/camera_info" to="camera1/rgb/camera_info" />
   	 	<param name="marker_pattern" type="string" value="$(find ar_pose)/data/4x4/4x4_97.patt"/>
    		<param name="marker_width" type="double" value="96.0"/>
    		<param name="marker_center_x" type="double" value="0.0"/>
    		<param name="marker_center_y" type="double" value="0.0"/>
    		<param name="threshold" type="int" value="120"/>
    		<param name="use_history" type="bool" value="true"/>
		<param name="marker_frame" value="marker" />
		<param name="publish_tf" value="false" />
		<remap from="ar_pose_marker" to="ar_pose/cam1/cp_5" />

		<param name="rgb_frame_id" value="camera_left" />
  	</node>

	<node name="ar_pose_c2_n5" pkg="ar_pose" type="ar_single" respawn="false"
   	 output="screen">

		<remap from="camera/image_raw" to="camera2/rgb/image" />
		<remap from="camera/camera_info" to="camera2/rgb/camera_info" />
   	 	<param name="marker_pattern" type="string" value="$(find ar_pose)/data/4x4/4x4_97.patt"/>
    		<param name="marker_width" type="double" value="96.0"/>
    		<param name="marker_center_x" type="double" value="0.0"/>
    		<param name="marker_center_y" type="double" value="0.0"/>
    		<param name="threshold" type="int" value="120"/>
    		<param name="use_history" type="bool" value="true"/>
		<param name="marker_frame" value="ar_marker" />
		<param name="publish_tf" value="false" />
		<remap from="ar_pose_marker" to="ar_pose/cam2/cp_5" />

		<param name="rgb_frame_id" value="camera_right" />
  	</node>

		<node name="app_msys_node" pkg="pos_detector" type="app_msys_node"
   	 output="screen" launch-prefix="$(arg launch_prefix)">
	</node>



</launch>
